### PR TITLE
[Reviewer: Andy] Integrate ENT logs with communication monitor

### LINKED ISSUE #317: ENT logs often spam
There are situations in which ENT logs on failures that could occur quite frequently - see https://github.com/Metaswitch/sprout/issues/973 for one such example.

The individual conditions are sensible, so I suspect we need to suppress ENT logs when they occur too frequently - e.g. generating one log when it first happens and then a summary log after some period (and nothing else).

### PR DESCRIPTION
**Pull request by eleanor-merry**
Andy, can you review this fix to add ENT logs to the communication monitor (so that the per call ENT logs can be dropped).

Fixes #317

### CODE REVIEW

File: src/communicationmonitor.cpp [hunk 1]
```diff
@@ -75,39 +76,43 @@ void CommunicationMonitor::track_communication_changes(unsigned long now_ms)
...
-                  succeeded, failed);
-        if ((succeeded == 0) && (failed != 0))
+        CL_CM_CONNECTION_ERRORED.log(_sender.c_str(),
+                                     _receiver.c_str());
+
+        if ((_alarm != NULL) && (!_alarm->alarmed()))
         {
           TRC_STATUS("Setting alarm %d", _alarm->index());
           _alarm->set();
+          _error_state = true;
```
Comment (bossmc) [inline src/communicationmonitor.cpp]: If we don't have an alarm, we'll drop the log every time round the loop.  Is this deliberate?

File: src/communicationmonitor.cpp [hunk 2]
```diff
@@ -75,39 +76,43 @@ void CommunicationMonitor::track_communication_changes(unsigned long now_ms)
...
-      else
+      else if ((_error_state) && (succeeded != 0))
       {
-        // A communication alarm is currently set so see if it needs to be
-        // cleared. This will be the case if at lease one successful comm
-        // was reported over the interval.
-        TRC_DEBUG("Alarm currently set - successful attempts %d",
-                  succeeded);
-        if (succeeded != 0)
+        CL_CM_CONNECTION_CLEARED.log(_sender.c_str(),
```
Comment (bossmc) [inline src/communicationmonitor.cpp]: If there's no alarm, this will never be hit (since we never enter error state).

**Comment by bossmc** [comment]
Generally looks good, I think there's an issue with uses where we don't have an alarm object to work with.

File: src/communicationmonitor.cpp (hunk 1 above)
Comment (eleanor-merry) [inline src/communicationmonitor.cpp]: It's not - I'll move it out of the if

**Pull request merged**